storage/storage_image.go

```diff
@@ -556,7 +556,8 @@ func (s *storageImageDestination) Commit() error {
 			return errors.Errorf("error applying blob %q: content not found", blob.Digest)
 		}
 		// Build the new layer using the diff, regardless of where it came from.
-		layer, _, err := s.imageRef.transport.store.PutLayer(id, lastLayer, nil, "", false, diff)
+		options := (*storage.LayerOptions)(nil)
+		layer, _, err := s.imageRef.transport.store.PutLayer(id, lastLayer, nil, "", false, options, diff)
```

Comment:
Why not just `…, false, nil, diff)`?

Or if you wanted to make it easy to add fields in the future, `options := &storage.LayerOptions{}`.  Setting the typed `nil` is a surprising mix of the two ideas which is more complex than either of them.

Reply:
I thought having the `nil` typed would make it more explicit in showing what we try to achieve.

I've changed it to untyped `nil` now -> 1efad7d255b3e1f0250beccbd8d3dc2608e02365.